src/protocols/protocol_address_31402.cpp

```diff
@@ -63,9 +62,7 @@ void protocol_address_31402::start()
     protocol_events::start(BIND1(handle_stop, _1));
 
     if (!self_.addresses().empty())
-    {
         SEND2(self_, handle_send, _1, self_.command);
-    }
 
```

Comment:
Macros should be braced as they may have multiple lines, resulting in insidious bugs.

Reply:
The macro sounds buggy then, but ok ...

```
#define MACRO do {
 // whatever a macro does
} while(0)
```

Reply:
It's not a problem with the macro, it's a best practice that avoids hard-to-see bugs. We try to avoid macros in non-test code because they are opaque and hard to debug, but when used they should be braced. We make an exception for the logging macros, since they are so common.

Reply:
Ah, I thought macros we're mostly for code obfuscation and brevity :slightly_smiling_face:

Reply:
Brevity isn't always a good thing.
```apl
life←{↑1 ω∨.∧3 4=+/,¯1 0 1∘.⊖¯1 0 1∘.⌽⊂ω}
```

Reply:
It was a joke ... I've worked in perl ;-)